client/cmd.go

```diff
@@ -373,3 +378,40 @@ func SetCmdClientContext(cmd *cobra.Command, clientCtx Context) error {
 
 	return nil
 }
+
+func GetViperFromCmd(cmd *cobra.Command) *viper.Viper {
+	value := cmd.Context().Value(corectx.ViperContextKey)
+	v, ok := value.(*viper.Viper)
+	if !ok {
+		return viper.New()
+	}
+	return v
+}
+
+func GetConfigFromCmd(cmd *cobra.Command) *cmtcfg.Config {
+	v := cmd.Context().Value(corectx.ViperContextKey)
+	viper, ok := v.(*viper.Viper)
+	if !ok {
+		return cmtcfg.DefaultConfig()
+	}
+	return GetConfigFromViper(viper)
+}
+
+func GetLoggerFromCmd(cmd *cobra.Command) log.Logger {
+	v := cmd.Context().Value(corectx.LoggerContextKey)
+	logger, ok := v.(log.Logger)
+	if !ok {
+		return log.NewLogger(os.Stdout)
+	}
+	return logger
+}
+
+func GetConfigFromViper(v *viper.Viper) *cmtcfg.Config {
```

Comment:
I am not sure this works in its current form. Intercept config never read the config.toml in viper

Reply:
I think in `interceptConfigs`, viper read both cmtConfig & appConfig by 
```go
rootViper.SetConfigType("toml")
rootViper.SetConfigName("config")
rootViper.AddConfigPath(configPath)

if err := rootViper.ReadInConfig()
```

and
 
```go
rootViper.SetConfigType("toml")
rootViper.SetConfigName("app")
rootViper.AddConfigPath(configPath)

if err := rootViper.MergeInConfig()
```

Also for tests, there is a step for reading config into viper in `genutil.TrackConfig`, basically same as above.

Reply:
Missed it, thanks!